core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -391,7 +406,9 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
             }
             new UpdateMetadataRequest.Broker(broker.id, endPoints.asJava, broker.rack.orNull)
           }
-          new UpdateMetadataRequest(version, controllerId, controllerEpoch, partitionStates.asJava, liveBrokers.asJava)
+          new UpdateMetadataRequest.Builder(
+              controllerId, controllerEpoch, partitionStates.asJava, liveBrokers.asJava).
+              setVersion(version)
```

Comment:
Looks like the code for both `Builder(...)` is the same, so maybe we can just do the `liveBrokers` bit in the `if/else`.